Windows HostProcess Container Documentation (#28413)
kubernetes/website: content/en/docs/concepts/security/pod-security-standards.md

				</ul>
			</td>
		</tr>
		<tr>
			<td style="white-space: nowrap">Windows HostProcess</td>
			<td>

> 				<p>Windows pods offer the ability to run <a href="/docs/tasks/configure-pod-container/create-hostprocess-pod">HostProcess containers</a> which enables privileged access to the Windows node. As is similar to  privileged containers this must be disallowed in the baseline policy. </p>

Reviewer: ```suggestion
				<p>Running Windows pods as <a href="/docs/tasks/configure-pod-container/create-hostprocess-pod">HostProcess containers</a> enables privileged access to the Windows node and should be disallowed.</p>
```

				<p><strong>Restricted Fields</strong></p>
				<ul>
					<li><code>spec.securityContext.windowsOptions.hostProcess</code></li>

> 					<li><code>spec.containers[*].securityContext.windowsOptions.hostProcess</code></li>

Reviewer: also

```
spec.initContainers[*].securityContext.windowsOptions.hostProcess
spec.ephemeralContainers[*].securityContext.windowsOptions.hostProcess
```

				</ul>
				<p><strong>Allowed Values</strong></p>
				<ul>

> 					<li>Undefined/nil</li>

Reviewer: ```suggestion
					<li>Undefined/null</li>
```

				</ul>
			</td>
		</tr>
		<tr>

> 			<td style="white-space: nowrap">Windows HostProcess</td>

Reviewer 2: This looks like a duplacet of the `HostProcess` row added above. Am I missing something?

Author: Nope it's a duplicate, removing it - thanks!

> ### What profiles should I apply to my Windows Pods?

Reviewer 3: Let's mention that HostProcess is alpha here, in case readers infer that it's a stable feature.

Windows in Kubernetes has some limitations and differentiators from standard Linux-based

> workloads. Specifically, a many of the Pod SecurityContext fields [have no effect on

Author: Yeah it's a typo, and also yes because there are some security context fields which apply to HostProcess

Windows](/docs/setup/production-environment/windows/intro-windows-in-kubernetes/#v1-podsecuritycontext). As
such, no standardized Pod Security profiles currently exist.

> Windows HostProcess containers can be enabled for the privileged profile, but are explicitly blocked for baseline.
> Windows pods _may_ be broken by the restricted profile, which requires setting linux-specific 
> settings (such as seccomp profile and disallow privilege escalation). If the Kubelet and/or 
> container runtime choose to ignore these linux-specific values at runtime, then Windows 
> pods should still be allowed under the restricted profile, although the profile will not 
> add additional enforcement over baseline (for Windows).

Reviewer 3: ```suggestion
If you apply the restricted profile for a Windows pod, this **may** have an impact on the pod
at runtime. The restricted profile requires enforcing Linux-specific restrictions (such as seccomp
profile, and disallowing privilege escalation). If the kubelet and / or its container runtime ignore
these Linux-specific values, then the Windows pod should still work normally within the restricted
profile. However, the lack of enforcement means that there is no additional restriction, for Pods
that use Windows containers, compared to the baseline profile.
```
?
